ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveRelDistribution.java

```diff
@@ -82,7 +83,14 @@ public RelDistribution apply(TargetMapping mapping) {
     }
     List<Integer> newKeys = new ArrayList<>(keys.size());
     for (Integer key : keys) {
-      newKeys.add(mapping.getTargetOpt(key));
+      // Instead of this inner for loop newKeys.add(mapping.getTargetOpt(key)); should be called but not all the
```

Comment:
nit. Can we add the mapping to a map and do the lookup, instead of having a nested loop?

Reply:
fixed